deepmd/pt/model/descriptor/sezm_nn/triton/so2_rotation.py

```
@@ -0,0 +1,1715 @@
+    strictly increasing in ``mmax`` (``lmax+1``, ``3*lmax+1``, ``5*lmax-1``, ...),
+    so ``Dm == 3*lmax+1`` uniquely pins ``mmax == 1``; combined with the model's
+    canonical ``build_m_major_index`` ordering this fully determines the layout.
+    """
+    dim_full = int(dim_full)
+    root = math.isqrt(dim_full)
+    if root * root != dim_full:
+        return -1
+    lmax = root - 1
+    try:
+        numel = int(coeff_index.shape[0])
+    except Exception:  # pragma: no cover - exotic shape proxies
```

Ruff (0.15.15)

[warning] Do not catch blind exception: `Exception`

(BLE001)